coil-compose-core/src/commonMain/kotlin/coil3/compose/AsyncImagePainter.kt

```diff
@@ -192,6 +192,13 @@ class AsyncImagePainter internal constructor(
     private var alpha: Float by mutableFloatStateOf(DefaultAlpha)
     private var colorFilter: ColorFilter? by mutableStateOf(null)
 
+    internal lateinit var scope: CoroutineScope
+    private var rememberJob: Job? = null
+        set(value) {
+            field?.cancel()
+            field = value
+        }
```

Comment:
@colinrtwhite I think here's a slight bug that causes the image to start loading later than necessary.

Because you're not checking for `if(value == field) return`, the `onRemembered()` call is first called manually and then by the Compose framework, which means that there will be two requests.